### PR TITLE
Added ReadInt32 for decoder

### PR DESCRIPTION
**Pull request by yzprofile**

### CODE REVIEW

File: decoder.go [hunk 1]
```diff
@@ -144,6 +144,11 @@ func (d *Decoder) Decode(v interface{}) (err error) {
 		n, err = d.ReadInt16()
 		rv.SetInt(int64(n))
 		return
+	case *int32:
```
Comment (maoueh) [inline decoder.go]: Going to merge, it's all good. Just curious, where it failing to decode a `int32`?

File: decoder_test.go [hunk 2]
```diff
@@ -153,6 +153,26 @@ func TestDecoder_Uint32(t *testing.T) {
 	assert.Equal(t, 0, d.remaining())
 }
 
+func TestDecoder_Int32(t *testing.T) {
+
+	buf := new(bytes.Buffer)
+	enc := NewEncoder(buf)
+	enc.writeInt32(int32(342))
```
Comment (maoueh) [inline decoder_test.go]: In fact, can you add a negative test like in [Line #116](https://github.com/eoscanada/eos-go/pull/66/files#diff-d477aef9bf305a8d89d0973a5b775b8aR116) so we cover both positive and negative numbers.

Once it's changed, I'm going to merge your work.

File: decoder.go (hunk 1 above)
Comment (yzprofile) [reply decoder.go]: in parsing table content.

**Comment by maoueh** [comment]
I've made the change and merged into master (https://github.com/eoscanada/eos-go/commit/690d9cf11be6e59a4be3a97c7aa4bcb7728e37da). Closing.

**Pull request closed without merging**